Fatal linker error LNK1181 on Windows

I'm trying to run `ui_tests` on Windows, but I keep getting the linker error "fatal error LNK1181: cannot open input file 'windows.0.48.5.lib'". I tried both `ui_tests` version `0.10` and `0.17`, but with no luck. The same tests work on Ubuntu and macOS.

To reproduce:
* Take the pull request https://github.com/duchess-rs/duchess/pull/111
* Run `cargo test --manifest-path=test-crates/Cargo.toml -p duchess-java-tests`

In particular, the `ui.rs` that launches `ui_tests` is this one: https://github.com/duchess-rs/duchess/blob/f5919288f212d9c60697a4c8d3b7bfc2d3e2cbb3/test-crates/duchess-java-tests/tests/ui.rs

Output:
```text
tests\ui\examples\log_local_ref_and_two_calls.rs:
  Error: no message
  Error: you likely need to bless the tests
  Error: Error: linking with `link.exe` failed: exit code: 1181
  pass test got exit code: 1, but expected 0
  
  actual output differed from expected
  Execute `"D:\\a\\duchess\\duchess\\test-crates\\target\\debug\\deps\\ui-bf3232b29b03ce5a.exe" --bless` to update `tests\ui\examples\log_local_ref_and_two_calls.stderr` to the actual output
  --- tests\ui\examples\log_local_ref_and_two_calls.stderr
  +++ <stderr output>
  +error: linking with `link.exe` failed: exit code: 1181
  +  |
  +  = note: "C:\\Program Files\\Microsoft Visual Studio\\2022\\Enterprise\\VC\\Tools\\MSVC\\14.36.32532\\bin\\HostX64\\x64\/link.exe" "/NOLOGO" "C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\rustcQXzJN9\/symbols.o" "..\\target\\ui\/log_local_ref_and_two_calls.log_local_ref_and_two_calls.76471adb22c5b38f-cgu.0.rcgu.o" "..\\target\\ui\/log_local_ref_and_two_calls.log_local_ref_and_two_calls.76471adb22c5b38f-cgu.1.rcgu.o" "..\\target\\ui\/log_local_ref_and_two_calls.log_local_ref_and_two_calls.76471adb22c5b38f-cgu.10.rcgu.o" "..\\target\\ui\/log_local_ref_and_two_calls.log_local_ref_and_two_calls.76471adb22c5b38f-cgu.11.rcgu.o" "..\\target\\ui\/log_local_ref_and_two_calls.log_local_ref_and_two_calls.76471adb22c5b38f-cgu.12.rcgu.o" "..\\target\\ui\/log_local_ref_and_two_calls.log_local_ref_and_two_calls.76471adb22c5b38f-cgu.13.rcgu.o" "..\\target\\ui\/log_local_ref_and_two_calls.log_local_ref_and_two_calls.76471adb22c5b38f-cgu.14.rcgu.o" "..\\target\\ui\/log_local_ref_and_two_calls.log_local_ref_and_two_calls.76471adb22c5b38f-cgu.15.rcgu.o" "..\\target\\ui\/log_local_ref_and_two_calls.log_local_ref_and_two_calls.76471adb22c5b38f-cgu.2.rcgu.o" "..\\target\\ui\/log_local_ref_and_two_calls.log_local_ref_and_two_calls.76471adb22c5b38f-cgu.3.rcgu.o" "..\\target\\ui\/log_local_ref_and_two_calls.log_local_ref_and_two_calls.76471adb22c5b38f-cgu.4.rcgu.o" "..\\target\\ui\/log_local_ref_and_two_calls.log_local_ref_and_two_calls.76471adb22c5b38f-cgu.5.rcgu.o" "..\\target\\ui\/log_local_ref_and_two_calls.log_local_ref_and_two_calls.76471adb22c5b38f-cgu.6.rcgu.o" "..\\target\\ui\/log_local_ref_and_two_calls.log_local_ref_and_two_calls.76471adb22c5b38f-cgu.7.rcgu.o" "..\\target\\ui\/log_local_ref_and_two_calls.log_local_ref_and_two_calls.76471adb22c5b38f-cgu.8.rcgu.o" "..\\target\\ui\/log_local_ref_and_two_calls.log_local_ref_and_two_calls.76471adb22c5b38f-cgu.9.rcgu.o" "..\\target\\ui\/log_local_ref_and_two_calls.324tu3cr00hcoivw.rcgu.o" "/LIBPATH:D:\\a\\duchess\\duchess\\test-crates\\duchess-java-tests\\..\\target\\ui\\debug\\deps" "/LIBPATH:D:\\a\\duchess\\duchess\\test-crates\\duchess-java-tests\\..\\target\\ui\\x86_64-pc-windows-msvc\\debug\\deps" "/LIBPATH:C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib" "D:\\a\\duchess\\duchess\\test-crates\\target\\ui\\x86_64-pc-windows-msvc\\debug\\deps\/libduchess-41e91d22ca22a63f.rlib" "D:\\a\\duchess\\duchess\\test-crates\\target\\ui\\x86_64-pc-windows-msvc\\debug\\deps\/libcesu8-4aa70e52252aab05.rlib" "D:\\a\\duchess\\duchess\\test-crates\\target\\ui\\x86_64-pc-windows-msvc\\debug\\deps\/libjava_locator-10ac43da87ffde02.rlib" "D:\\a\\duchess\\duchess\\test-crates\\target\\ui\\x86_64-pc-windows-msvc\\debug\\deps\/liblazy_static-2c1780ce1f1c49f4.rlib" "D:\\a\\duchess\\duchess\\test-crates\\target\\ui\\x86_64-pc-windows-msvc\\debug\\deps\/libglob-990d2ccf93aaa33e.rlib" "D:\\a\\duchess\\duchess\\test-crates\\target\\ui\\x86_64-pc-windows-msvc\\debug\\deps\/libtracing-b053a9666d3ad529.rlib" "D:\\a\\duchess\\duchess\\test-crates\\target\\ui\\x86_64-pc-windows-msvc\\debug\\deps\/libcfg_if-6ff079a5faaf2937.rlib" "D:\\a\\duchess\\duchess\\test-crates\\target\\ui\\x86_64-pc-windows-msvc\\debug\\deps\/libpin_project_lite-2978b65e242f8cfb.rlib" "D:\\a\\duchess\\duchess\\test-crates\\target\\ui\\x86_64-pc-windows-msvc\\debug\\deps\/libtracing_core-5027c1360c57a43e.rlib" "D:\\a\\duchess\\duchess\\test-crates\\target\\ui\\x86_64-pc-windows-msvc\\debug\\deps\/libjni_sys-3f48a63c4dd8d548.rlib" "D:\\a\\duchess\\duchess\\test-crates\\target\\ui\\x86_64-pc-windows-msvc\\debug\\deps\/liblibloading-a21ff566406b1c80.rlib" "D:\\a\\duchess\\duchess\\test-crates\\target\\ui\\x86_64-pc-windows-msvc\\debug\\deps\/libwindows_sys-1fb863ca2ba2b90c.rlib" "D:\\a\\duchess\\duchess\\test-crates\\target\\ui\\x86_64-pc-windows-msvc\\debug\\deps\/libwindows_targets-a40b71bbee77917b.rlib" "D:\\a\\duchess\\duchess\\test-crates\\target\\ui\\x86_64-pc-windows-msvc\\debug\\deps\/libonce_cell-393a315680f7bc46.rlib" "D:\\a\\duchess\\duchess\\test-crates\\target\\ui\\x86_64-pc-windows-msvc\\debug\\deps\/libthiserror-8b187e744d0c5a5b.rlib" "C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\/libstd-4e8f46f7d0c95848.rlib" "C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\/libpanic_unwind-5db344dab6eb1d2b.rlib" "C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\/librustc_demangle-74e09b816cbd5555.rlib" "C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\/libstd_detect-0e78b5195f1cb27a.rlib" "C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\/libhashbrown-bfdbaec42e428240.rlib" "C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\/librustc_std_workspace_alloc-ece31b1fa03921a4.rlib" "C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\/libminiz_oxide-b1cda3daaedb751f.rlib" "C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\/libadler-fcb0daf3e92ec962.rlib" "C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\/libunwind-943c8b115715a79a.rlib" "C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\/libcfg_if-86ba8a5505ea3939.rlib" "C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\/liblibc-5e13ccbf89e39d2e.rlib" "C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\/liballoc-150c9edec7571387.rlib" "C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\/librustc_std_workspace_core-cd5722df944967ab.rlib" "C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\/libcore-f671e54f90f2e08c.rlib" "C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\/libcompiler_builtins-0157311a6d6a27aa.rlib" "windows.0.48.5.lib" "kernel32.lib" "advapi32.lib" "bcrypt.lib" "kernel32.lib" "ntdll.lib" "userenv.lib" "ws2_32.lib" "kernel32.lib" "msvcrt.lib" "/NXCOMPAT" "/LIBPATH:C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib" "/OUT:..\\target\\ui\/log_local_ref_and_two_calls.exe" "/OPT:REF,NOICF" "/DEBUG" "/NATVIS:C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\/intrinsic.natvis" "/NATVIS:C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\/liballoc.natvis" "/NATVIS:C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\/libcore.natvis" "/NATVIS:C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\/libstd.natvis"
  +  = note: LINK : fatal error LNK1181: cannot open input file 'windows.0.48.5.lib'
  +          
  +
  +error: aborting due to previous error
  +
  
  
  There were 1 unmatched diagnostics that occurred outside the testfile and had no pattern
      Error: linking with `link.exe` failed: exit code: 1181
  
  full stderr:
  error: linking with `link.exe` failed: exit code: 1181
    |
    = note: "C:\\Program Files\\Microsoft Visual Studio\\2022\\Enterprise\\VC\\Tools\\MSVC\\14.36.32532\\bin\\HostX64\\x64\\link.exe" "/NOLOGO" "C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\rustcQXzJN9\\symbols.o" "..\\target\\ui\\log_local_ref_and_two_calls.log_local_ref_and_two_calls.76471adb22c5b38f-cgu.0.rcgu.o" "..\\target\\ui\\log_local_ref_and_two_calls.log_local_ref_and_two_calls.76471adb22c5b38f-cgu.1.rcgu.o" "..\\target\\ui\\log_local_ref_and_two_calls.log_local_ref_and_two_calls.76471adb22c5b38f-cgu.10.rcgu.o" "..\\target\\ui\\log_local_ref_and_two_calls.log_local_ref_and_two_calls.76471adb22c5b38f-cgu.11.rcgu.o" "..\\target\\ui\\log_local_ref_and_two_calls.log_local_ref_and_two_calls.76471adb22c5b38f-cgu.12.rcgu.o" "..\\target\\ui\\log_local_ref_and_two_calls.log_local_ref_and_two_calls.76471adb22c5b38f-cgu.13.rcgu.o" "..\\target\\ui\\log_local_ref_and_two_calls.log_local_ref_and_two_calls.76471adb22c5b38f-cgu.14.rcgu.o" "..\\target\\ui\\log_local_ref_and_two_calls.log_local_ref_and_two_calls.76471adb22c5b38f-cgu.15.rcgu.o" "..\\target\\ui\\log_local_ref_and_two_calls.log_local_ref_and_two_calls.76471adb22c5b38f-cgu.2.rcgu.o" "..\\target\\ui\\log_local_ref_and_two_calls.log_local_ref_and_two_calls.76471adb22c5b38f-cgu.3.rcgu.o" "..\\target\\ui\\log_local_ref_and_two_calls.log_local_ref_and_two_calls.76471adb22c5b38f-cgu.4.rcgu.o" "..\\target\\ui\\log_local_ref_and_two_calls.log_local_ref_and_two_calls.76471adb22c5b38f-cgu.5.rcgu.o" "..\\target\\ui\\log_local_ref_and_two_calls.log_local_ref_and_two_calls.76471adb22c5b38f-cgu.6.rcgu.o" "..\\target\\ui\\log_local_ref_and_two_calls.log_local_ref_and_two_calls.76471adb22c5b38f-cgu.7.rcgu.o" "..\\target\\ui\\log_local_ref_and_two_calls.log_local_ref_and_two_calls.76471adb22c5b38f-cgu.8.rcgu.o" "..\\target\\ui\\log_local_ref_and_two_calls.log_local_ref_and_two_calls.76471adb22c5b38f-cgu.9.rcgu.o" "..\\target\\ui\\log_local_ref_and_two_calls.324tu3cr00hcoivw.rcgu.o" "/LIBPATH:D:\\a\\duchess\\duchess\\test-crates\\duchess-java-tests\\..\\target\\ui\\debug\\deps" "/LIBPATH:D:\\a\\duchess\\duchess\\test-crates\\duchess-java-tests\\..\\target\\ui\\x86_64-pc-windows-msvc\\debug\\deps" "/LIBPATH:C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib" "D:\\a\\duchess\\duchess\\test-crates\\target\\ui\\x86_64-pc-windows-msvc\\debug\\deps\\libduchess-41e91d22ca22a63f.rlib" "D:\\a\\duchess\\duchess\\test-crates\\target\\ui\\x86_64-pc-windows-msvc\\debug\\deps\\libcesu8-4aa70e52252aab05.rlib" "D:\\a\\duchess\\duchess\\test-crates\\target\\ui\\x86_64-pc-windows-msvc\\debug\\deps\\libjava_locator-10ac43da87ffde02.rlib" "D:\\a\\duchess\\duchess\\test-crates\\target\\ui\\x86_64-pc-windows-msvc\\debug\\deps\\liblazy_static-2c1780ce1f1c49f4.rlib" "D:\\a\\duchess\\duchess\\test-crates\\target\\ui\\x86_64-pc-windows-msvc\\debug\\deps\\libglob-990d2ccf93aaa33e.rlib" "D:\\a\\duchess\\duchess\\test-crates\\target\\ui\\x86_64-pc-windows-msvc\\debug\\deps\\libtracing-b053a9666d3ad529.rlib" "D:\\a\\duchess\\duchess\\test-crates\\target\\ui\\x86_64-pc-windows-msvc\\debug\\deps\\libcfg_if-6ff079a5faaf2937.rlib" "D:\\a\\duchess\\duchess\\test-crates\\target\\ui\\x86_64-pc-windows-msvc\\debug\\deps\\libpin_project_lite-2978b65e242f8cfb.rlib" "D:\\a\\duchess\\duchess\\test-crates\\target\\ui\\x86_64-pc-windows-msvc\\debug\\deps\\libtracing_core-5027c1360c57a43e.rlib" "D:\\a\\duchess\\duchess\\test-crates\\target\\ui\\x86_64-pc-windows-msvc\\debug\\deps\\libjni_sys-3f48a63c4dd8d548.rlib" "D:\\a\\duchess\\duchess\\test-crates\\target\\ui\\x86_64-pc-windows-msvc\\debug\\deps\\liblibloading-a21ff566406b1c80.rlib" "D:\\a\\duchess\\duchess\\test-crates\\target\\ui\\x86_64-pc-windows-msvc\\debug\\deps\\libwindows_sys-1fb863ca2ba2b90c.rlib" "D:\\a\\duchess\\duchess\\test-crates\\target\\ui\\x86_64-pc-windows-msvc\\debug\\deps\\libwindows_targets-a40b71bbee77917b.rlib" "D:\\a\\duchess\\duchess\\test-crates\\target\\ui\\x86_64-pc-windows-msvc\\debug\\deps\\libonce_cell-393a315680f7bc46.rlib" "D:\\a\\duchess\\duchess\\test-crates\\target\\ui\\x86_64-pc-windows-msvc\\debug\\deps\\libthiserror-8b187e744d0c5a5b.rlib" "C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libstd-4e8f46f7d0c95848.rlib" "C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libpanic_unwind-5db344dab6eb1d2b.rlib" "C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\librustc_demangle-74e09b816cbd5555.rlib" "C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libstd_detect-0e78b5195f1cb27a.rlib" "C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libhashbrown-bfdbaec42e428240.rlib" "C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\librustc_std_workspace_alloc-ece31b1fa03921a4.rlib" "C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libminiz_oxide-b1cda3daaedb751f.rlib" "C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libadler-fcb0daf3e92ec962.rlib" "C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libunwind-943c8b115715a79a.rlib" "C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libcfg_if-86ba8a5505ea3939.rlib" "C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\liblibc-5e13ccbf89e39d2e.rlib" "C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\liballoc-150c9edec7571387.rlib" "C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\librustc_std_workspace_core-cd5722df944967ab.rlib" "C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libcore-f671e54f90f2e08c.rlib" "C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libcompiler_builtins-0157311a6d6a27aa.rlib" "windows.0.48.5.lib" "kernel32.lib" "advapi32.lib" "bcrypt.lib" "kernel32.lib" "ntdll.lib" "userenv.lib" "ws2_32.lib" "kernel32.lib" "msvcrt.lib" "/NXCOMPAT" "/LIBPATH:C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib" "/OUT:..\\target\\ui\\log_local_ref_and_two_calls.exe" "/OPT:REF,NOICF" "/DEBUG" "/NATVIS:C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\intrinsic.natvis" "/NATVIS:C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\liballoc.natvis" "/NATVIS:C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\libcore.natvis" "/NATVIS:C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\libstd.natvis"
    = note: LINK : fatal error LNK1181: cannot open input file 'windows.0.48.5.lib'
            
  
  error: aborting due to previous error
  
  
  

tests\ui\examples\log_one_big_call.rs FAILED:
command: "rustc" "--error-format=json" "--extern" "duchess=D:\\a\\duchess\\duchess\\test-crates\\duchess-java-tests\\..\\target\\ui\\x86_64-pc-windows-msvc\\debug\\deps\\libduchess-41e91d22ca22a63f.rlib" "--extern" "duchess=D:\\a\\duchess\\duchess\\test-crates\\duchess-java-tests\\..\\target\\ui\\x86_64-pc-windows-msvc\\debug\\deps\\libduchess-41e91d22ca22a63f.rmeta" "--extern" "thiserror=D:\\a\\duchess\\duchess\\test-crates\\duchess-java-tests\\..\\target\\ui\\x86_64-pc-windows-msvc\\debug\\deps\\libthiserror-8b187e744d0c5a5b.rlib" "--extern" "thiserror=D:\\a\\duchess\\duchess\\test-crates\\duchess-java-tests\\..\\target\\ui\\x86_64-pc-windows-msvc\\debug\\deps\\libthiserror-8b187e744d0c5a5b.rmeta" "-L" "D:\\a\\duchess\\duchess\\test-crates\\duchess-java-tests\\..\\target\\ui\\debug\\deps" "-L" "D:\\a\\duchess\\duchess\\test-crates\\duchess-java-tests\\..\\target\\ui\\x86_64-pc-windows-msvc\\debug\\deps" "--out-dir" "..\\target\\ui" "tests\\ui\\examples\\log_one_big_call.rs" "--edition" "2021"
```

Have you already seen anything like this?

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -17,6 +17,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Fixed
 
+* folders and libraries linked by build scripts were ignored, causing linker failures
+
 ### Changed
 
 * removed `Revisioned::no_rustfix` in favor of turning that into a rustc-specific custom flag

--- a/src/dependencies.rs
+++ b/src/dependencies.rs
@@ -1,6 +1,6 @@
 //! Use `cargo` to build dependencies and make them available in your tests
 
-use cargo_metadata::{camino::Utf8PathBuf, DependencyKind};
+use cargo_metadata::{camino::Utf8PathBuf, BuildScript, DependencyKind};
 use cargo_platform::Cfg;
 use color_eyre::eyre::{bail, eyre, Result};
 use std::{
@@ -25,6 +25,8 @@ pub struct Dependencies {
     /// All paths that must be imported with `-L dependency=`. This is for
     /// finding proc macros run on the host and dependencies for the target.
     pub import_paths: Vec<PathBuf>,
+    /// Unnamed dependencies that build scripts asked us to link
+    pub import_libs: Vec<PathBuf>,
     /// The name as chosen in the `Cargo.toml` and its corresponding rmeta file.
     pub dependencies: Vec<(String, Vec<Utf8PathBuf>)>,
 }
@@ -95,31 +97,43 @@ fn build_dependencies_inner(config: &Config, info: &DependencyBuilder) -> Result
     let artifact_output = output.stdout;
     let artifact_output = String::from_utf8(artifact_output)?;
     let mut import_paths: HashSet<PathBuf> = HashSet::new();
+    let mut import_libs: HashSet<PathBuf> = HashSet::new();
     let mut artifacts = HashMap::new();
     'artifact: for line in artifact_output.lines() {
         let Ok(message) = serde_json::from_str::<cargo_metadata::Message>(line) else {
             continue;
         };
-        if let cargo_metadata::Message::CompilerArtifact(artifact) = message {
-            for ctype in &artifact.target.crate_types {
-                match ctype.as_str() {
-                    "proc-macro" | "lib" => {}
-                    _ => continue 'artifact,
+        match message {
+            cargo_metadata::Message::CompilerArtifact(artifact) => {
+                for ctype in &artifact.target.crate_types {
+                    match ctype.as_str() {
+                        "proc-macro" | "lib" => {}
+                        _ => continue 'artifact,
+                    }
+                }
+                for filename in &artifact.filenames {
+                    import_paths.insert(filename.parent().unwrap().into());
+                }
+                let package_id = artifact.package_id;
+                if let Some(prev) = artifacts.insert(package_id.clone(), Ok(artifact.filenames)) {
+                    artifacts.insert(
+                        package_id.clone(),
+                        Err(format!(
+                            "{prev:#?} vs {:#?} ({:?})",
+                            artifacts[&package_id], artifact.target.crate_types
+                        )),
+                    );
                 }
             }
-            for filename in &artifact.filenames {
-                import_paths.insert(filename.parent().unwrap().into());
+            cargo_metadata::Message::BuildScriptExecuted(BuildScript {
+                linked_libs,
+                linked_paths,
+                ..
+            }) => {
+                import_paths.extend(linked_paths.into_iter().map(Into::into));
+                import_libs.extend(linked_libs.into_iter().map(Into::into));
             }
-            let package_id = artifact.package_id;
-            if let Some(prev) = artifacts.insert(package_id.clone(), Ok(artifact.filenames)) {
-                artifacts.insert(
-                    package_id.clone(),
-                    Err(format!(
-                        "{prev:#?} vs {:#?} ({:?})",
-                        artifacts[&package_id], artifact.target.crate_types
-                    )),
-                );
-            }
+            _ => {}
         }
     }
 
@@ -204,9 +218,11 @@ fn build_dependencies_inner(config: &Config, info: &DependencyBuilder) -> Result
             })
             .collect::<Result<Vec<_>>>()?;
         let import_paths = import_paths.into_iter().collect();
+        let import_libs = import_libs.into_iter().collect();
         return Ok(Dependencies {
             dependencies,
             import_paths,
+            import_libs,
         });
     }
 
@@ -286,6 +302,10 @@ pub fn build_dependencies(config: &Config, info: &DependencyBuilder) -> Result<V
     }
     for import_path in dependencies.import_paths {
         args.push("-L".into());
+        args.push(import_path.into());
+    }
+    for import_path in dependencies.import_libs {
+        args.push("-l".into());
         args.push(import_path.into());
     }
     Ok(args)

